axelrod/_strategy_utils.py

```diff
@@ -26,18 +26,15 @@ def detect_cycle(history, min_size=1, max_size=12, offset=0):
         The amount of history to skip initially
     """
     history_tail = history[offset:]
-    max_ = min(len(history_tail) // 2, max_size)
-    for i in range(min_size, max_):
+    new_max_size = min(len(history_tail) // 2, max_size)
+    for i in range(min_size, new_max_size + 1):
         has_cycle = True
         cycle = tuple(history_tail[:i])
         for j, elem in enumerate(history_tail):
             if elem != cycle[j % len(cycle)]:
                 has_cycle = False
                 break
-        if has_cycle and (j == len(history_tail) - 1):
```

Comment:
i couldn't figure out what j==len(history_tail) -1 did, and i couldn't write a test that needed it.  i'm not sure if there's some test that i missed.